IndexOf(Char, StringComparison) is used instead of IndexOf(Char, Int32), resulting in build errors.

### Steps to reproduce

Add the following to a project file or targets file that gets included when building:

```xml
  <PropertyGroup>
    <Revision>$([System.DateTime]::Now.ToString(yyyyMMdd)).99</Revision>
    <TheIndex>$(Revision.IndexOf('.', 10))</TheIndex>
  </PropertyGroup>
```

### Expected behavior
The project can build successfully and IndexOf(Char, Int32) is invoked on Revision.

### Actual behavior
An error is produced:
```
error MSB4184: The expression ""20200519.99".IndexOf(., 10)" cannot be evaluated. The string comparison type passed in is currently not supported. Parameter name: comparisonType
```

It appears that it's only attempting to use IndexOf(Char, StringComparison) and 10 isn't
a valid value for the StringComparison enum. However, it should still be using IndexOf(Char, Int32)
### Environment data
`msbuild /version` output:

Microsoft (R) Build Engine version 16.7.0-preview-20265-07+31a1a34cb for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

16.7.0.26507

OS info:
Windows - 10.0.18362

If applicable, version of the tool that invokes MSBuild (Visual Studio, dotnet CLI, etc):
Occurs opening the project in VS or when trying to build the project from a VS command prompt.

--- a/src/Build/Evaluation/Expander.cs
+++ b/src/Build/Evaluation/Expander.cs
@@ -3537,14 +3537,6 @@ namespace Microsoft.Build.Evaluation
                             return true;
                         }
                     }
-                    else if (string.Equals(_methodMethodName, nameof(string.IndexOf), StringComparison.OrdinalIgnoreCase))
-                    {
-                        if (TryGetArgs(args, out string arg0, out StringComparison arg1))
-                        {
-                            returnVal = text.IndexOf(arg0, arg1);
-                            return true;
-                        }
-                    }
                     else if (string.Equals(_methodMethodName, nameof(string.IndexOfAny), StringComparison.OrdinalIgnoreCase))
                     {
                         if (TryGetArg(args, out string arg0))
@@ -3565,10 +3557,20 @@ namespace Microsoft.Build.Evaluation
                             returnVal = text.LastIndexOf(arg0, startIndex);
                             return true;
                         }
-                        else if (TryGetArgs(args, out arg0, out StringComparison arg1))
+                        else if (TryGetArgs(args, out arg0, out string arg1))
                         {
-                            returnVal = text.LastIndexOf(arg0, arg1);
-                            return true;
+                            string comparisonType = arg1;
+
+                            // Allow fully-qualified enum, e.g. "System.StringComparison.OrdinalIgnoreCase"
+                            if (comparisonType.Contains("."))
+                            {
+                                comparisonType = arg1.Replace("System.StringComparison.", "").Replace("StringComparison.", "");
+                            }
+                            if (Enum.TryParse<StringComparison>(comparisonType, out StringComparison comparison))
+                            {
+                                returnVal = text.LastIndexOf(arg0, comparison);
+                                return true;
+                            }
                         }
                     }
                     else if (string.Equals(_methodMethodName, nameof(string.Length), StringComparison.OrdinalIgnoreCase))
@@ -4238,33 +4240,6 @@ namespace Microsoft.Build.Evaluation
 
                 arg0 = args[0] as string;
                 return arg0 != null;
-            }
-
-            private static bool TryGetArgs(object[] args, out string arg0, out StringComparison arg1)
-            {
-                if (args.Length != 2)
-                {
-                    arg0 = null;
-                    arg1 = default;
-
-                    return false;
-                }
-
-                arg0 = args[0] as string;
-
-                if (arg0 == null || !(args[1] is string comparisonType))
-                {
-                    arg1 = default;
-                    return false;
-                }
-
-                // Allow fully-qualified enum, e.g. "System.StringComparison.OrdinalIgnoreCase"
-                if (comparisonType.Contains('.'))
-                {
-                    comparisonType = comparisonType.Replace("System.StringComparison.", "").Replace("StringComparison.", "");
-                }
-
-                return Enum.TryParse<StringComparison>(comparisonType, out arg1);
             }
 
             private static bool TryGetArgs(object[] args, out int arg0, out int arg1)

